src/utils/item.ts

```diff
@@ -130,3 +130,88 @@ export const paginationContentFilter = (
   items
     .filter((i) => i.type !== ItemType.FOLDER)
     .filter((i) => !i.settings?.isPinned);
+
+// interface Node {
+//   children: ;
+// }
+
+interface Tree {
+  [nodeId: string]: DiscriminatedItem[];
+}
+
+const createMapTree = (data: DiscriminatedItem[]) => {
+  const childrenTreeMap: Tree = {};
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < data.length; i++) {
+    const ele = data[i];
+    const parents = getParentsIdsFromPath(ele.path, { ignoreSelf: true });
+    const lastParent = parents[parents.length - 1];
+    if (lastParent) {
+      if (childrenTreeMap[lastParent]) {
+        childrenTreeMap[lastParent] = [...childrenTreeMap[lastParent], ele];
+      } else {
+        childrenTreeMap[lastParent] = [ele];
+      }
+    }
+  }
+  return childrenTreeMap;
+};
+
+type ItemWithChildren = DiscriminatedItem & {
+  children: ItemWithChildren[];
+};
+
+interface TreeNode {
+  [nodeId: string]: ItemWithChildren;
+}
+
+// handle item children tree
+const buildItemsTree = (data: DiscriminatedItem[]) => {
+  const tree: TreeNode = {};
+  if (data.length === 1) {
+    tree[data[0].id] = { ...data[0], children: [] };
+  }
+  const mapTree: Tree = createMapTree(data);
+  const keys = Object.keys(mapTree);
+  const allChildren: DiscriminatedItem[] = keys.reduce(
+    (acc: DiscriminatedItem[], key: string) => {
+      const node = mapTree[key as keyof Tree];
+      if (node && node.length) {
+        acc.push(...node);
+      }
+      return acc;
+    },
+    [],
+  );
+
+  const rootKeys = keys.filter(
+    (key) => !allChildren.find((ele) => ele.id === key),
+  );
+
+  const buildTree = (nodeId: string) => {
```

Comment:
What about my previous comment? Sorry if you didn't check my comments yet, I thought you needed a review.